src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/bucketize_ref.cl

```diff
@@ -0,0 +1,64 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "include/batch_headers/data_types.cl"
+#include "include/batch_headers/fetch_data.cl"
+
+KERNEL(bucketize_ref)(const __global INPUT0_TYPE* data,
+                    const __global INPUT1_TYPE* buckets,
+                    __global OUTPUT_TYPE* output
+#if HAS_FUSED_OPS_DECLS
+                           , FUSED_OPS_DECLS
+#endif
+)
+{
+    const uint batch = get_global_id(0);
+    const uint feature = get_global_id(1);
+
+#if OUTPUT_DIMS == 5
+    const uint z = ((uint)get_global_id(2) / OUTPUT_SIZE_X) / OUTPUT_SIZE_Y;
+    const uint y = ((uint)get_global_id(2) / OUTPUT_SIZE_X) % OUTPUT_SIZE_Y;
+    const uint x = (uint)get_global_id(2) % OUTPUT_SIZE_X;
+#else
+    const uint z = 0;
+    const uint y = (uint)get_global_id(2) / OUTPUT_SIZE_X;
+    const uint x = (uint)get_global_id(2) % OUTPUT_SIZE_X;
+#endif
+
+    const uint input_index = INPUT0_GET_INDEX(batch, feature, z, y, x);
+    for (int i = 0; i < BUCKETS_SHAPE; i++) {
+        if (WITH_RIGHT_BOUND) {
+            if (data[input_index] == buckets[i]) {
+                const uint output_index = OUTPUT_GET_INDEX(batch, feature, z, y, x);
+#if HAS_FUSED_OPS
```

Comment:
It looks like that currently your kernel doesn't really support fused ops => you can remove this code from here.